src/java.base/share/classes/java/text/DigitList.java

```diff
@@ -788,23 +788,23 @@ public String toString() {
         if (isZero()) {
             return "0";
         }
-        StringBuffer buf = getStringBuffer();
-        buf.append("0.");
-        buf.append(digits, 0, count);
-        buf.append("x10^");
-        buf.append(decimalAt);
-        return buf.toString();
+        StringBuilder temp = getStringBuilder();
+        temp.append("0.");
+        temp.append(digits, 0, count);
+        temp.append("x10^");
+        temp.append(decimalAt);
+        return temp.toString();
```

Comment:
This could use chained method calls and avoid the local variable completely.

Reply:
Thanks for pointing that out, made the switch there and in the getDouble() method as well.